qemu/tests/qemu_guest_agent.py

```diff
@@ -1721,6 +1721,101 @@ def gagent_check_path_fsfreeze_hook(self, test, params, env):
         else:
             test.fail("The default path of fsfreeze-hook doesn't match with expectation.")
 
+    @error_context.context_aware
+    def gagent_check_fsfreeze_hook_script(self, test, params, env):
+        """
+        During fsfreeze,verify fsfreeze hook script works.
+
+        Steps:
+        1) Check fsfreeze hook related files.
+        2) Check fsfreeze hook path set in qemu-ga config file.
+        3) Fsfreeze hook should be with '-x' permission for all users.
+        4) Verify agent service is using the fsfreeze hook.
+        5) Create a simple user script in hook scripts path.
+        6) Get fsfreeze hook log file.
+        7) Issue freeze & thaw cmds.
+        8) Check fsfreeze hook logs.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        """
+
+        def log_check(action):
+            msg = "testing %s:%s" % (user_script_path, action)
+            hook_log = session.cmd_output("cat %s" % log_path)
+            if msg not in hook_log.strip().split('\n')[-2]:
+                test.fail("Fsfreeze hook test failed\nthe fsfreeze"
+                          " hook log is %s." % hook_log)
+
+        session = self._get_session(self.params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Checking fsfreeze hook related scripts.",
+                              logging.info)
+        cmd_get_hook_files = "rpm -ql qemu-guest-agent |grep fsfreeze-hook"
+        hook_files = session.cmd_output(cmd_get_hook_files)
+
+        if len(hook_files.strip().split()) != 4:
```

Comment:
@xiagao why !=4?

Reply:
Because there are 4 files related to hook.

Reply:
Acked by: Xiangchun Fu xfu@redhat.com

Reply:
I cannot see any point checking the fsfreeze hook script path here, since it could be specified with any position with `--fsfreeze-hook`